src/renderers/dom/shared/dangerousStyleValue.js

```diff
@@ -48,7 +51,27 @@ function dangerousStyleValue(name, value) {
   }
 
   if (typeof value === 'string') {
-    value = value.trim();
+    if (__DEV__) {
+      if (component) {
+        var owner = component._currentElement._owner;
+        var componentName = owner ? owner.getName() : component._currentElement.type;
+        if (!styleWarnings[componentName]) {
+          styleWarnings[componentName] = {};
+        }
+        var warnings = styleWarnings[componentName];
+        if (!warnings[name]) {
+          warnings[name] = true;
+          warning(
+            false,
+            '`%s` passed a numeric string value for CSS property `%s` ' +
+            'which will be treated as a unitless number.',
+            componentName,
+            name
+          );
+        }
+      }
+    }
+    return value.trim();
```

Comment:
Okay. Should I just revert the original change (i.e. remove the return statement and adjust the tests and docs) then?